langlib/lang.object/src/main/ballerina/object.bal

```diff
@@ -13,3 +13,69 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+
+# Distinct Iterable type.
+# An object can make itself iterable by using `*object:Iterable;`,
+# and then defining an `iterator` method.
+public type Iterable distinct object {
+    # Create a new iterator.
+    #
+    # + return - a new iterator object
+    public function iterator() returns object {
+        public isolated function next() returns record {| any|error value; |}|error?;
+    };
+};
+
+// moved here from __internal
+
+# Integer range expression is represented using `IntRange` object.
+#
+# + iStart - start expression of range expression
+# + iEnd - second expression on range expression
+# + iCurrent - current cursor
+public class __IntRange {
```

Comment:
These aren't in the spec, right? Probably why we had it in the __internal module in the first place.

Reply:
Yes, then internal will depend on `lang.object` vice versa.

Reply:
The problem with this is that the users will start seeing this in suggestions, API docs, etc. Is the dependency required for `Iterable`?

Reply:
We use `createIntRange` in desugar. `__IntRange` should be a subtype of `Iterable`. So Iterable should be available when we create `__IntRange`. Does creating interanal.bal inside lang.object module work?

Reply:
Is `createIntRange` used in langlibs too? As long as is lang.object is also available at the time this check is done we can work around this, right?